docs/firmware/golioth-firmware-sdk/light-db-state/guide-lightdb-observe.md

```diff
@@ -0,0 +1,100 @@
+---
+title: LightDB Observe
+sidebar_position: 3
+---
+
+Use a LightDB observe function to receive an update whenever a specific path
+value changes on the cloud. The observe request works in a similar way to the
```

Comment:
It could be useful to describe how an observe call is different from an asynchronous get (mostly for those unfamiliar with CoAP).

Reply:
I guess I'm also unfamiliar with CoAP. What is the difference?

My understanding is that an async get is initiated by the device (pull) but an observation is initiated by the cloud (push).

Reply:
The difference is that a `get()` is still an immediate operation - you're pulling the value to get it "now-ish" - the asynchronous part is just so that the CPU can do something else while it waits for the network operation. That's why there's still a timeout, and why that timeout is relatively short (a second or so). A get is also a one-shot operation - the callback will only be called one time for each call to `get()`.

Whereas an observe is more about setting a long-lived observation on a path. There's no expectation that the callback will be called anytime soon, and it can be called many times from just one call to `observe()`.

I think it can be helpful to think about different domains - at the network level, a `get()` is always a synchronous operation and will return a response, and an `observe()` is always asynchronous. But `_sync()` vs `_async()` is about what's happening on the CPU. Network operations take time, and so from the perspective of what the CPU is doing, an `_async()` call allows the CPU to do something else instead of waiting for the response to `get()`.

Reply:
I made an update to discuss difference between async get and observe. I also opened [an internal issue](https://github.com/golioth/firmware-issue-tracker/issues/193) to track adding information about when observations should be registered.